corehq/apps/sso/tests/test_tasks.py

```diff
@@ -133,3 +139,103 @@ def tearDown(self):
     def tearDownClass(cls):
         cls.account.delete()
         super().tearDownClass()
+
+
+class TestAutoDeactivationTask(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.account = generator.get_billing_account_for_idp()
+        cls.account.enterprise_admin_emails = ['test@vaultwax.com']
+        cls.account.save()
+        cls.domain = Domain.get_or_create_with_name("vaultwax-001", is_active=True)
+
+        enterprise_plan = accounting_generator.subscribable_plan_version(edition=SoftwarePlanEdition.ENTERPRISE)
+        accounting_generator.generate_domain_subscription(
+            cls.account,
+            cls.domain,
+            date_start=datetime.date.today(),
+            date_end=None,
+            plan_version=enterprise_plan,
+            is_active=True,
+        )
+
+        cls.idp = generator.create_idp('vaultwax', cls.account)
+        cls.idp.enable_user_deactivation = True
+        cls.idp.idp_type = IdentityProviderType.AZURE_AD
+        cls.idp.save()
+        cls.email_domain = AuthenticatedEmailDomain.objects.create(
+            email_domain='vaultwax.com',
+            identity_provider=cls.idp,
+        )
+
+    def setUp(self):
+        super().setUp()
+        self.web_user_a = self._create_web_user('a@vaultwax.com')
+        self.web_user_b = self._create_web_user('b@vaultwax.com')
+        self.web_user_c = self._create_web_user('c@vaultwax.com')
+        self.web_user_d = self._create_web_user('d@vaultwax.com')
+
+    @patch('corehq.apps.sso.models.IdentityProvider.get_all_members_of_the_idp')
+    def test_user_deactivation_logic(self, mock_get_all_members_of_the_idp):
```

Comment:
nit: would be nice to see a more specific test name. What is it about the deactivation logic that you are testing? It looks like you are testing that web_user_c is deactivated if no longer part of the idp? So maybe `test_user_is_deactivated_if_not_member_of_idp`

Reply:
Good suggestion! Thank you! Refactored here: a1940ef363250d2c4d7b5ca6b1ee708fb4081e81